pkg/sync/sync_context.go

```
@@ -1075,6 +1118,15 @@ func (sc *syncContext) applyObject(t *syncTask, dryRun, validate bool) (common.R
 	shouldReplace := sc.replace || resourceutil.HasAnnotationOption(t.targetObj, common.AnnotationSyncOptions, common.SyncOptionReplace)
 	force := sc.force || resourceutil.HasAnnotationOption(t.targetObj, common.AnnotationSyncOptions, common.SyncOptionForce)
 	serverSideApply := sc.shouldUseServerSideApply(t.targetObj, dryRun)
+
+	// Check if we need to perform client-side apply migration for server-side apply
+	if serverSideApply && !dryRun && sc.needsClientSideApplyMigration(t.liveObj) {
+		err = sc.performClientSideApplyMigration(t.targetObj)
+		if err != nil {
+			sc.log.WithValues("resource", kubeutil.GetResourceKey(t.targetObj)).Error(err, "Failed to perform client-side apply migration")
```

Is this failure mode loud enough? Should we instead fail the operation?

---

I decided here not to fail the operation as i didn't want to prevent the entire Sync from happening in case the client-side apply fails. What do you think?

---

I think we should return this error to the user instead of "hiding" it in the logs. Make sure that the error propagates in the sync context.